spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2406,19 +2406,19 @@ object QueryPlanSerde extends Logging with CometExprShim {
 
           // TODO this could be optimized more to stop walking the tree on hitting
           //  certain operators such as join or aggregate which will copy batches
-          def containsNativeCometScan(plan: SparkPlan): Boolean = {
+          def containsNonDataFusionScan(plan: SparkPlan): Boolean = {
             plan match {
-              case w: CometScanWrapper => containsNativeCometScan(w.originalPlan)
-              case scan: CometScanExec => scan.scanImpl == CometConf.SCAN_NATIVE_COMET
+              case _: CometScanWrapper => true
```

Comment:
Changed it to return true as native scan exec is not wrapped in CometScanWrapper

Reply:
and when not scan datafusion